src/main/java/io/quarkus/gizmo2/creator/ModifierFlag.java

```diff
@@ -0,0 +1,107 @@
+package io.quarkus.gizmo2.creator;
+
+import static io.github.dmlloyd.classfile.ClassFile.*;
+import static io.quarkus.gizmo2.creator.ModifierLocation.*;
+
+import java.util.EnumSet;
+
+/**
+ * A modifier for a type or member.
+ */
+public enum ModifierFlag implements Modifier {
```

Comment:
I'd personally love this to be called just `Modifier`, but I have no idea what current `Modifier` should be called. `AbstractModifier` is not great, but I cannot think of a better name ATM.

Reply:
Alternatively, looking at both `Access` and `ModifierFlag`, I don't think there's too big of a value in keeping them separate. We could just merge them.

Reply:
Yeah, agree that separating them is useful. But it brings the naming issue; `ModifierFlag` is really not a good name.

Reply:
How about we call this `Modifier` and what's now called `Modifier` would be called `Flag`? Then, we would have 2 kinds of flags: modifiers and access levels. The field to store both kinds is already called `flags`.

Reply:
Well a "flag" is a binary indicator (this definition goes back decades), whereas both access levels and the boolean modifiers are considered modifiers according to the JVMS, so that seems backwards to me in that regard since an access level is not a flag but it is a modifier.

Reply:
Technically access levels are also flags then, because there's one bit position for each of them. The JVMS actually uses the term flag for access levels too.

However, if we really need to differentiate access levels and other modifiers, I think I'd be fine with calling this `BooleanModifier`, `BinaryModifier`, `SimpleModifier`, `PropertyModifier` or something like that. If we could avoid using the term "flag" in this context completely, that would be a nice bonus :-)

Reply:
Access levels are encoded as flags, but they aren't flags, because the *four* access levels are represented by *three* flags which are mutually exclusive. They're flags in the JVM, but not in the language.

I don't know what's wrong with the word "flag" as it has a clear meaning and historical relevance in computing as well.

Reply:
The only thing wrong about "flags" is that in the JLS and JVMS and practically everywhere in Java, they are a synonym of "modifiers".

Now, practically speaking, I feel bad about stalling this PR for so long. Since this is the only objection unresolved, I'm gonna cave in and approve this PR so you can merge. Let's get this in and perfect it later on :-)